### PR TITLE
Cypress e2e test

### PR DESCRIPTION
**Pull request by grant-davidson**
This test fails under windows as the symlink public/gif.worker.js to

   ../node_modules/gif.js/dist/gif.worker.js

doesn't work.

### CODE REVIEW

**Comment by dai-shi** [comment]
wow, this looks nice. does it run on ci?

btw, you are not the same person as @raphaelbs, right? just in case.

**Comment by raphaelbs** [comment]
> btw, you are not the same person as @raphaelbs, right? just in case.

lol!

It is great that @grant-davidson could start this.
I'm [migrating this project](https://github.com/raphaelbs/infire-delivery/pull/196) that uses Cy to use the Cy dashboard. The idea is to apply the learnings here.

Suggestions that I can already provide:

For @dai-shi:
- Create a project on https://dashboard.cypress.io/
- Setup the `CYPRESS_RECORD_KEY` as a secret on the repo

For @grant-davidson 
- Using GitHub actions as done here: [.github/workflows/cypress.yml](.github/workflows/cypress.yml)
- Adding the `projectId` on the cypress.json file

By doing so, we can have [neat comments on the PRs](https://github.com/raphaelbs/infire-delivery/pull/196#issuecomment-803459033).

**Comment by grant-davidson** [comment]
Thanks, It should run on ci.  I have something similar running on bitbucket
pipelines.
I'll need to do some tidying up as eslint is currently not happy with the
cy object not being declared.

No, I'm not @raphaelbs <https://github.com/raphaelbs>

**Comment by dai-shi** [comment]
Oh, I'm so happy to see there are **two** people interested in working on this project.

@grant-davidson Would you be able to join my discord server https://discord.gg/MrQdmzd ? We have a channel #excalidraw-claymate.

**Comment by dai-shi** [comment]
> Setup the `CYPRESS_RECORD_KEY` as a secret on the repo

Done.

```json
{
    "projectId": "8759xt"
}
```

**Comment by dai-shi** [comment]
Looks good? Can @raphaelbs review?

image

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    runs-on: ubuntu-16.04
+    steps:
+      - uses: actions/checkout@v1
+      - uses: cypress-io/github-action@v2
+        with:
+          browser: chrome
+          headless: true
+          start: yarn start
+        env:
+          CYPRESS_CI: true
```
Comment (raphaelbs) [inline .github/workflows/cypress.yml:14]:
```suggestion
          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}

```

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: Cypress
+on: [push,pull_request]
+jobs:
+  cypress-run:
+    runs-on: ubuntu-16.04
+    steps:
+      - uses: actions/checkout@v1
+      - uses: cypress-io/github-action@v2
+        with:
+          browser: chrome
```
Comment (raphaelbs) [inline .github/workflows/cypress.yml:10]:
```suggestion
          record: true
          browser: chrome
```

**Review by raphaelbs — COMMENTED** [review]
Adding the `CYPRESS_RECORD_KEY` should publish the report on Cy dashboard and comment the status here.

Also, @dai-shi, I couldn't access the [project URL](https://dashboard.cypress.io/projects/8759xt/runs). Is it that one, right?

Maybe its visibility is private.
You can check that by going to the `project settings` on Cy dashboard:

image

**Comment by dai-shi** [comment]
> Also, @dai-shi, I couldn't access the [project URL](https://dashboard.cypress.io/projects/8759xt/runs). Is it that one, right?
> 
> Maybe its visibility is private.

You are right. It was private. I changed it public now. But, it's not yet recorded?

image

**Comment by raphaelbs** [comment]
> But, it's not yet recorded?

It will be once the workflow file has been changed

**Comment by grant-davidson** [comment]
The action log includes:

`Warning: It looks like you are trying to record this run from a forked PR.

The 'Record Key' is missing. Your CI provider is likely not passing private environment variables to builds from forks.

These results will not be recorded.`

**Comment by dai-shi** [comment]
Okay, let's merge this and see how it goes. Thanks for your work.

**Pull request merged**